File: FRM_NamePicker.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace RandomNamePicker
{
    public partial class FRM_NamePicker : Form
    {
        public FRM_NamePicker()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }

        private void button1_Click(object sender, EventArgs e)
        {

            if(TXT_SelectCount.Text != "" && TXT_Entries.Text != "")
            {

                TXT_Selected.Text = GenerateName(Convert.ToInt32(TXT_SelectCount.Text), TXT_Entries.Lines);
            }
            else
            {
                MessageBox.Show("Missing Information");
            }



            
        }
        public static string GenerateName(int len, string[] Names)
        {
            Random r = new Random();

            string Name = "";
            int b = 0;
            while (b < len)
            {
                Name += Names[r.Next(Names.Length)] + Environment.NewLine;
                b++;
            }

            return Name;


        }

    }
}

File: FRM_NamePicker.vb
Imports System
Imports System.Windows.Forms

Namespace RandomNamePicker
    Public Partial Class FRM_NamePicker
        Inherits Form

        Public Sub New()
            Me.InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim ctr = Convert.ToInt32("2")

            While ctr <> 1
                Dim names As String = RandName()
                ctr += 1
            End While
        End Sub

        Public Function RandName() As String
            Dim names = {"Name 1", "Name 2", "Name 3", "Name 4"}
            Dim rand As Random = New Random()
            Dim index = rand.Next(names.Length)
            Return $"{names(index)}"
        End Function
    End Class
End Namespace
